tss-esapi-sys/build.rs

```diff
@@ -1,114 +1,487 @@
+        _tss2_sys: Library,
+        tss2_esys: Library,
+        tss2_tctildr: Library,
+        tss2_mu: Library,
+        tss2_tcti_tbs: Option<Library>,
+    }
+
+    impl Installation {
+        /// Probes the system for an installation.
+        pub fn probe(with_header_files: bool) -> Self {
+            let install_path = Installation::installation_path_from_env_var();
+            Installation {
+                _tss2_sys: Library::probe_required(
+                    "tss2-sys",
+                    install_path.as_ref(),
+                    with_header_files,
+                    false,
+                ),
+                tss2_esys: Library::probe_required(
+                    "tss2-esys",
+                    install_path.as_ref(),
+                    with_header_files,
+                    true,
+                ),
+                tss2_tctildr: Library::probe_required(
+                    "tss2-tctildr",
+                    install_path.as_ref(),
+                    with_header_files,
+                    false,
+                ),
+                tss2_mu: Library::probe_required(
+                    "tss2-mu",
+                    install_path.as_ref(),
+                    with_header_files,
+                    false,
+                ),
+                tss2_tcti_tbs: Library::probe_optional(
+                    "tss2-tcti-tbs",
+                    install_path.as_ref(),
+                    with_header_files,
+                ),
+            }
+        }
+
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "generate-bindings")] {
+
+                /// Generates bindings for the Installation.
+                pub fn generate_bindings(&self, esapi_out: &Path) {
+                    self.bindgen_builder()
+                        .generate()
+                        .expect("Unable to generate bindings to TSS2 ESYS APIs.")
+                        .write_to_file(esapi_out)
+                        .expect("Couldn't write ESYS bindings!");
+                }
+
+                /// The bindgen builder to use.
+                fn bindgen_builder(&self) -> bindgen::Builder {
+                    let mut builder = bindgen::Builder::default()
+                        .size_t_is_usize(false)
+                        .clang_arg(self.tss2_esys.include_dir_arg())
+                        .clang_arg(self.tss2_tctildr.include_dir_arg())
+                        .clang_arg(self.tss2_mu.include_dir_arg())
+                        .formatter(bindgen::Formatter::Rustfmt)
+                        .header(self.tss2_esys.header_file_arg())
+                        .header(self.tss2_tctildr.header_file_arg())
+                        .header(self.tss2_mu.header_file_arg())
+                        //See this issue: https://github.com/parallaxsecond/rust-cryptoki/issues/12
+                        .generate_comments(false)
+                        .blocklist_type("max_align_t")
+                        // Needed for windows
```

Comment:
What was the issue here?

Reply:
When you compile under windows ( I used msvc 2022) you get a lot of errors of structs that it some how thinks are needed but not available probably due to it importing "windows.h" or something. So I just blocked everything that caused an error.

Reply:
I meant when you generate the bindings on a windows machine.

Reply:
> When you compile under windows ( I used msvc 2022) you get a lot of errors of structs that it some how thinks are needed but not available probably due to it importing "windows.h" or something.

I wonder if [`WIN32_LEAN_AND_MEAN`](https://devblogs.microsoft.com/oldnewthing/20091130-00/?p=15863) would help here... (note that I'm no C/C++ dev so this may be wildly wrong)...